tutorial_pl/README.md

```diff
@@ -0,0 +1,152 @@
+
+```
+nazwa_zadania             - głowny folder paczki
+├── doc/tag.pdf           - treść (najlepiej też z plikiem tag.tex, kod źródłowy treści w latexie)
+├── in/tag*.in            - testy (pliki wejściowe)
+├── out/tag*.out          - testy (pliki wyjściowe)
+├── prog/tag*.cpp         - wzorcówki / rozwiązania (też niekoniecznie poprawne)
+├── check/checker.cpp     - checkerka (zazwyczaj niepotrzebna)
+├── utils/                - miejsce na pomocnicze programy/skrypty, typu generatorka (ignorowany folder)
+├── Simfile               - plik konfiguracyjny dla sim'a
+└── Sipfile               - plik konfiguracyjny dla sip'a
+```
+
+# Treść
+
+Jeżeli posiadasz już plik pdf z treścią zadania, umieść go w folderze doc/. Jeżeli nie, sip kompiluje kod źródłowy latexa komendą `sip doc`.
+TODO: tutorial do pisania treści
+
+# Wzorcówki / rozwiązania
+
+W folderze prog/ umieść wszystkie rozwiązania do zadania. Bardzo zalecane jest napisanie rozwiązań wzorcowych, rozwiązań wolnych (brutów) i rozwiązań dające niepoprawne odpowiedzi. Sip akceptuje rozwiązania napisane w C++ (z rozszerzeniem .cpp) oraz w Pascalu (z rozszerzeniem .pas). Sip automatycznie traktuje rozwiązanie z najkrótszą nazwą jako rozwiązanie wzorcowe. W paczkach zadań Olimpiady Informatycznej można spotkać się ze zwyczajem, by rozwiązania zapisywać pod nazwami tag#.cpp, tags#.cpp, tagb#.cpp oznaczające kolejno rozwiązania poprawne, wolne i dające niepoprawne odpowiedzi (znak # oznacza dowolną liczbę), jednak sim nie wymaga takiego nazewnictwa plików. Przy dużej ilości rozwiązań rzeczywiście taki sposób jest sensowny, jednak zazwyczaj wystarczy nazywać pliki z krótkim opisem, np. tag_brute.cpp. 
+
+Rozwiązania powinny wczytywać ze standardowego wejścia (np. cin) i wypisywać na standardowe wyjście (np. cout). Bardzo zalecane jest umieszczenie komentarza na samej górze każdego kodu z opisem rozwiązania (czy jest poprawne, czy nie - jak nie to dlaczego nie), opisem złożoności i autorem rozwiązania.
+
+
+# Generatorka
+
+By wygenerować testy, potrzebna jest generatorka. Program ma przyjmować argumenty i na standardowe wyjście (np. cout) wypisać test. Za pomocą argumentów określa się rozmiar testu, itp. Przykładowa generatorka:
+
+``` cpp
+#include <bits/stdc++.h>
+using namespace std;
+using L = long long;
+
+mt19937_64 engine(chrono::system_clock::now().time_since_epoch().count());
+// losowy int z przedziału [a, b]: (więcej informacji na en.cppreference.com/w/cpp/numeric/random )
+L rd(L a, L b) {
+	return uniform_int_distribution<L>(a, b)(engine);
+}
+
+int main(int argc, char* argv[]) {
+	// argument 0 jest pomijany
+	constexpr int ilosc_argumentow = 2;
+	assert(argc == ilosc_argumentow + 1);
+
+	// zamienianie z tablicy charów na liczbę (więcej informacji na en.cppreference.com/w/cpp/string/byte/atoi )
+	int max_n = atoi(argv[1]),
+	    max_val = atoi(argv[2]);
+
+	int n = rd(1, max_n);
+	cout << n << '\n';
+	for(int i = 0; i < n; ++i)
+		cout << rd(-max_val, max_val) << ' ';
```

Comment:
Dobrze jest aby generatoka nie zostawiała białych znaków na końcu linii i aby output kończył się enterem - wydaje mi się, że taki format plików in jest "najbardziej wzorcowy" i tego wypadałoby się trzymać.